discussions-default/src/main/java/org/xwiki/contrib/discussions/internal/DefaultDiscussionContextService.java

```diff
@@ -96,12 +100,14 @@ public Optional<DiscussionContext> create(String applicationHint, String name, S
     @Override
     public List<DiscussionContext> findByDiscussionReference(DiscussionReference reference)
     {
-        return this.discussionContextStoreService.findByDiscussionReference(reference)
+        List<DiscussionContext> result = this.discussionContextStoreService.findByDiscussionReference(reference)
             .stream()
             .map(this::mapBaseObject)
             .filter(Optional::isPresent)
             .map(Optional::get)
             .collect(Collectors.toList());
+        result.forEach(discussionContext -> this.discussionContextMetadataStoreService.readMetadata(discussionContext));
```

Comment:
I think you can use `peek` after `.map(Optional::get)` instead (if you want to avoid defining an intermediate `result` variable).